packages/erxes-ui/src/components/richTextEditor/RichTextEditorContent/RichTextEditorContent.tsx

```diff
@@ -39,10 +39,32 @@ export const RichTextEditorContent = (props: IRichTextEditorContentProps) => {
     return value;
   }
 
+  const shouldShow = useCallback(
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **issue (complexity):** Consider encapsulating the `BubbleMenu` logic into a separate component or custom hook.

The introduction of `useCallback` and the complex logic within `shouldShow` significantly increases the cognitive load required to understand the conditions under which the `BubbleMenu` is displayed. This, combined with the added dependencies and conditional rendering logic, makes the component more complex and potentially harder to maintain.

Consider encapsulating the `BubbleMenu` logic, including the `shouldShow` function and its dependencies, into a separate component or custom hook. This would not only simplify the `RichTextEditorContent` component, making it more focused on its primary responsibilities, but also improve maintainability by isolating complex logic. Additionally, simplifying the conditional rendering and styling logic could further reduce complexity and enhance readability.